### PR TITLE
(MODULES-8432) refresh PA repo if the version is not in the local cache

### PR DESCRIPTION
**Pull request by mihaibuzgau**
This PR adds the functionality to refresh the local cache of the Suse package repository if the desired version of the Puppet Agent is not found.

### CODE REVIEW

File: manifests/osfamily/suse.pp [hunk 1]
```diff
@@ -96,6 +97,13 @@
             value   => $value,
           }
         }
+
+        exec { "refresh-${repo_name}":
```
Comment (ekinanp) [inline manifests/osfamily/suse.pp]: I think the [`ini_setting` resource should have a `before => Exec["refresh-${repo_name}"]`](https://github.com/puppetlabs/puppetlabs-puppet_agent/blob/master/manifests/osfamily/suse.pp#L91-L97) to make sure that the repo's created prior to refreshing it.

Comment (speedofdark) [reply manifests/osfamily/suse.pp]: Could you please point me at a reference to `before`? I can only find it in the context of an `insert_type`, and not as a standalone element. Thanks.

Comment (ekinanp) [reply manifests/osfamily/suse.pp]: https://puppet.com/docs/puppet/5.3/lang_relationships.html#syntax-relationship-metaparameters

**Comment by mihaibuzgau** [comment]
@ekinanp @speedofdark can I merge this ?

**Pull request merged**